recurring_contract/tests/test_base_contract.py

```diff
@@ -31,7 +31,8 @@ def setUp(self):
         property_account_payable = account_obj.search([
             ('type', '=', 'payable'),
             ('user_type', '=', account_type)]).ids[0]
-
+        category = self.env['res.partner.category'].create({
+            'name': 'sponsor'})

```

Comment:
This should not be in `test_base_contract` as there are no sponsors in this module. Instead, you could move this create in the setup of test of module `contract_compassion` and then: 
```
self.partners.write({'category_id': [(4, category.id)]})
````